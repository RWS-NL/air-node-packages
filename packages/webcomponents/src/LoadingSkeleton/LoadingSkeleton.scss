@import '../styles/vars';

@keyframes skeleton-keyframes {
  from {
    background-position: -200px 0;
  }

  to {
    background-position: calc(200px + 100%) 0;
  }
}

.skeleton {
  animation-duration: 1.2s;
  animation-iteration-count: infinite;
  animation-name: skeleton-keyframes;
  animation-timing-function: ease-in-out;
  background-color: $grey10;
  background-image: linear-gradient(90deg, $grey10, $grey20, $grey10);
  background-repeat: no-repeat;
  background-size: 200px 100%;
  border-radius: 4px;
  display: inline-block;
  line-height: 1;
  width: 100%;
}
