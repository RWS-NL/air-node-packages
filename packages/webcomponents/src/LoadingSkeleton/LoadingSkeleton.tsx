import { CSSProperties } from '@material-ui/styles';
import classnames from 'classnames';
import css from './LoadingSkeleton.scss';
import React, { FC, memo } from 'react';

export interface LoadingSkeletonProps {
  /**
   * Height of the skeleton
   * Useful when you don't want to adapt the skeleton to a text element but for instance a card
   * Also needed for the prop circle (see below).
   */
  height?: string | number;
  /**
   * Width of the skeleton
   * Useful when the skeleton is inside an inline element with no width of its own
   */
  width?: string | number;
  /**
   * Prop for making the skeleton look like a circle
   * for when you are creating a user card with a profile picture for instance
   */
  circle?: boolean;
  /** Any additional CSSProperties to pass to the component */
  style?: CSSProperties;
  /** Data-qa tag to apply to the search bar and input element */
  'data-qa'?: string;
  /** Custom CSS classes to pass to the button */
  customclasses?: string | string[];
}

/**
 * Constructs a loading skeleton to reserve space for loading data
 * @param props Props to pass to the loading skeleton
 * @example
 * <LoadingSkeleton
 *   data-qa='loading-skeleton'
 *   width={500} height={50}
 * />
 */
export const LoadingSkeleton: FC<LoadingSkeletonProps> = props => {
  const style: CSSProperties = {
    ...props.style,
    width: props.width,
    height: props.height,
    borderRadius: props.height && props.width && props.circle ? '50%' : '0%',
  };

  // Content is a zero-width-non-joiner
  return <span data-qa={props['data-qa']} className={classnames(css.skeleton, props.customclasses)} style={style}>&zwnj;</span>;
};

export default memo(LoadingSkeleton);