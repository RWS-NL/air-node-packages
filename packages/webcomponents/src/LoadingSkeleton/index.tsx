import { CSSProperties } from '@material-ui/styles';
import classnames from 'classnames';
import React, { memo } from 'react';
import css from './LoadingSkeleton.scss';

export interface LoadingSkeletonProps {
  /**
   * Height of the skeleton
   * Useful when you don't want to adapt the skeleton to a text element but for instance a card
   * Also needed for the prop circle (see below).
   */
  height?: string | number;
  /**
   * Width of the skeleton
   * Useful when the skeleton is inside an inline element with no width of its own
   */
  width?: string | number;
  /**
   * Prop for making the skeleton look like a circle
   * for when you are creating a user card with a profile picture for instance
   */
  circle?: boolean;
  /** Any additional CSSProperties to pass to the component */
  style?: CSSProperties;
  /** Data-qa tag to apply to the search bar and input element */
  'data-qa'?: string;
  /** Custom CSS classes to pass to the button */
  customclasses?: string | string[];
}

/**
 * Constructs a loading skeleton to reserve space for loading data
 * @param props Props to pass to the loading skeleton
 * @example
 * ```jsx
 * <LoadingSkeleton
 *   data-qa='loading-skeleton'
 *   width={500} height={50}
 * />
 * ```
 */
export const LoadingSkeleton = memo(
  ({ width, height, circle, style, 'data-qa': dataQa, ...props }: LoadingSkeletonProps) => {
    const cssStyle: CSSProperties = {
      ...style,
      width: width,
      height: height,
      borderRadius: height && width && circle ? '50%' : '0%'
    };

    // Content is a zero-width-non-joiner
    return (
      <span {...props} data-qa={dataQa} className={classnames(css.skeleton, props.customclasses)} style={cssStyle}>
        &zwnj;
      </span>
    );
  }
);

export default LoadingSkeleton;
