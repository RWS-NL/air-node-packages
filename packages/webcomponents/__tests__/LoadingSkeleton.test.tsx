import { shallow } from 'enzyme';
import LoadingSkeleton from '../src/LoadingSkeleton';
import React from 'react';

describe('Snapshot Testing', () => {
  test('Required Props', () => {
    const loadingSkeleton = shallow(<LoadingSkeleton />);
    expect(loadingSkeleton).toMatchSnapshot();
  });

  test('Optional Props', () => {
    const loadingSkeleton = shallow(<LoadingSkeleton height={200} width={200} />);
    expect(loadingSkeleton).toMatchSnapshot();
  });

  test('Circular Skeleton', () => {
    const loadingSkeleton = shallow(<LoadingSkeleton height={200} width={200} circle />);
    expect(loadingSkeleton).toMatchSnapshot();
  });
});
