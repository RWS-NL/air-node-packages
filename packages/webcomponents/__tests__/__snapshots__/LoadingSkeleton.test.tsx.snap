// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Circular Skeleton 1`] = `
<span
  className="skeleton"
  style={
    Object {
      "borderRadius": "50%",
      "height": 200,
      "width": 200,
    }
  }
>
  ‌
</span>
`;

exports[`Snapshot Testing Optional Props 1`] = `
<span
  className="skeleton"
  style={
    Object {
      "borderRadius": "0%",
      "height": 200,
      "width": 200,
    }
  }
>
  ‌
</span>
`;

exports[`Snapshot Testing Required Props 1`] = `
<span
  className="skeleton"
  style={
    Object {
      "borderRadius": "0%",
      "height": undefined,
      "width": undefined,
    }
  }
>
  ‌
</span>
`;
